README.md

```diff
@@ -15,33 +15,33 @@ When persisting this collection of values (Mysql, Elasticsearch), we use intensi
 
 ## Problems
 
-Do not that in theory, these problems should be highlited by the use of the profiler.
+Do note that, in theory these problems should be highlited by the use of the profiler.
 It's the role of the profile to determine if an optimization worths an improvement or not.
 
-Here, we are doing the opposite: it is to better understand the problem in the Blackfire profile.
+Here, we are doing the opposite: it is better to understand the problem in a Blackfire profile.
 
 ### Normalizer pattern
 
-The normalizer pattern iterates over all normalizers declared in the serializer, in order to find the good one for a given object.
+The normalizer pattern iterates over all normalizers declared in the serializer in order to find the good one for a given object (`supports` function).
 The more you have normalizers in the registry, slower it is to find the good normalizer.
 
-So, when normalizing value collection of 1000 products values and having 15 normalizers registered in the serializer, a lot of time is wasted at finding the normalizer for the product values. 
+So, when normalizing a value collection of 1000 products values and having 15 normalizers registered in the serializer, a lot of time is wasted at finding the normalizer for the product values. 
 
-Note: The complexity is O(m * n) where "m" is the number of product values and "n" the number of normalizers declared in the registry.
+Note: The complexity  of looping on it is O(m * n) where "m" is the number of product values and "n" the number of normalizers declared in the registry.
 
-The solution to avoid this complexity is to call directly the normalizer instead of calling the serializer.
+The solution to avoid this complexity is to call directly the normalizer instead of calling the serializer, or to have a map with supports function (only for class) and Normalizer themselves which is O(1).
```

Comment:
Actually, it's useless to have a map if you can inject it directly. I prefer to keep it as before.

Reply:
I do agree ;), I just wanted to add more explanation

Reply:
Yes, but it's already very verbose.